File: src/main/java/song/Verse.java
package song;

import java.util.Map;
import java.util.stream.Collectors;

public record Verse(
        Integer verse,
        Map<String, String> text) {

    public Map<String, String[]> getText() {
        return text.entrySet().stream()
                .collect(Collectors.toMap(
                        e -> e.getKey(),
                        e -> e.getValue().split("\n")));
    }
}

File: src/main/kotlin/com/scottscmo/model/song/Verse.kt
package com.scottscmo.model.song

class Verse() {
    lateinit var verse: String
    lateinit var text: Map<String, String>

    fun textLines(): Map<String, List<String>> {
        return this.text.map {
            it.key to it.value.split("\n")
        }.toMap()
    }
}
